str.go

```diff
@@ -0,0 +1,87 @@
+package merkletree
+
+// SignedTreeRoot represents a signed tree root, which is generated
+// at the beginning of every epoch.
+// Signed tree roots contain the current root node,
+// the current and previous epochs, the hash of the
+// previous STR, and its signature.
+// STR should be final
+type SignedTreeRoot struct {
+	treeRoot    *interiorNode
+	epoch       int64
+	prevEpoch   int64
+	prevStrHash []byte
+	sig         []byte
+	prev        *SignedTreeRoot
+}
+
+func (m *MerkleTree) generateSTR(ep int64, prevEp int64, prevHash []byte) *SignedTreeRoot {
+	bytesPreSig := getSTRBytesForSig(m, ep, prevEp, prevHash)
+	sig := m.scheme.Sign(bytesPreSig)
+
+	return &SignedTreeRoot{
+		treeRoot:    &(*m.root),
```

Comment:
Why do you dereference and immediately reference here again? Shouldn't that give you the same adress at the end? (like here: https://play.golang.org/p/wt-JIa-5w_)